core/src/test/scala/integration/kafka/api/ConsumerTopicCreationTest.scala

```diff
@@ -17,82 +17,88 @@
 
 package kafka.api
 
-import java.lang.{Boolean => JBoolean}
-import java.time.Duration
-import java.util
-import java.util.Collections
-
 import kafka.server.KafkaConfig
 import kafka.utils.TestUtils
 import org.apache.kafka.clients.admin.NewTopic
 import org.apache.kafka.clients.consumer.ConsumerConfig
 import org.apache.kafka.clients.producer.{ProducerConfig, ProducerRecord}
-import org.junit.Assert._
-import org.junit.Test
-import org.junit.runner.RunWith
-import org.junit.runners.Parameterized
-import org.junit.runners.Parameterized.Parameters
+import org.junit.jupiter.api.Assertions._
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.{Arguments, MethodSource}
+
+import java.lang.{Boolean => JBoolean}
+import java.time.Duration
+import java.util
+import java.util.Collections
 
 /**
  * Tests behavior of specifying auto topic creation configuration for the consumer and broker
  */
-@RunWith(value = classOf[Parameterized])
-class ConsumerTopicCreationTest(brokerAutoTopicCreationEnable: JBoolean, consumerAllowAutoCreateTopics: JBoolean) extends IntegrationTestHarness {
-  override protected def brokerCount: Int = 1
+class ConsumerTopicCreationTest {
+  @ParameterizedTest
+  @MethodSource(Array("parameters"))
+  def testAutoTopicCreation(brokerAutoTopicCreationEnable: JBoolean, consumerAllowAutoCreateTopics: JBoolean): Unit = {
+    val testCase = new ConsumerTopicCreationTest.TestCase(brokerAutoTopicCreationEnable, consumerAllowAutoCreateTopics)
+    testCase.setUp()
+    try testCase.test() finally testCase.tearDown()
+  }
+}
+
+object ConsumerTopicCreationTest {
+
+  class TestCase(brokerAutoTopicCreationEnable: JBoolean, consumerAllowAutoCreateTopics: JBoolean) extends IntegrationTestHarness {
+    private[this] val topic_1 = "topic-1"
+    private[this] val topic_2 = "topic-2"
+    private[this] val producerClientId = "ConsumerTestProducer"
+    private[this] val consumerClientId = "ConsumerTestConsumer"
```

Comment:
`private[this]` seems a bit unnecessary for a test, no? Seems like `private` is good enough.